include/libdnf/base/goal.hpp

```diff
@@ -256,9 +256,23 @@ class Goal {
     /// Also packages of the types specified in setting.group_package_types be installed.
     ///
     /// @param spec      A string describing the Goal group install request.
+    /// @param reason    Reason why the group is installed (USER/DEPENDENCY)
     /// @param settings  A structure to override default goal settings.
     void add_group_install(
-        const std::string & spec, const libdnf::GoalJobSettings & settings = libdnf::GoalJobSettings());
+        const std::string & spec,
+        const libdnf::transaction::TransactionItemReason reason,
+        const libdnf::GoalJobSettings & settings = libdnf::GoalJobSettings());
+
+    /// Add group remove request to the goal. The `spec` will be resolved to groups in the resolve() call.
+    /// Also packages not belonging to another group and not installed explicitly user-installed will get removed.
```

Comment:
Thanks @kontura. I've addressed all your comments.

Reply:
The build is failing now with:
```
/builddir/build/BUILD/dnf5-5.0.1-20221206144407.33.g42c83340/dnf5/commands/group/group_remove.cpp: In member function 'virtual void dnf5::GroupRemoveCommand::configure()':
/builddir/build/BUILD/dnf5-5.0.1-20221206144407.33.g42c83340/dnf5/commands/group/group_remove.cpp:43:13: error: 'class dnf5::Context' has no member named 'set_available_repos_load_flags'
   43 |     context.set_available_repos_load_flags(libdnf::repo::LoadFlags::PRIMARY | libdnf::repo::LoadFlags::COMPS);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Perhaps because https://github.com/rpm-software-management/dnf5/pull/123 was merged?
I think the new `group remove` command needs to be updated.

Reply:
Yeah, it's becouse the rebase. Should be fixed by now.

Reply:
Thanks!